From d588c746f4bdafc63131bb987e33b3882e460c26 Mon Sep 17 00:00:00 2001
From: Suyun <suyun@aosc.io>
Date: Mon, 14 Sep 2026 15:19:30 +0800
Subject: [PATCH] AOSCOS: Update Zydis with width marcos

Use width marcos instead of a hardcoded architectures list to prevent lacks of some architectures such as MIPS.

Ref: https://github.com/zyantific/zydis/commit/6372690e30389a94db65ece2d8a1f0a2310475ed
---
 .../zydis/amalgamated-dist/Zydis.c             | 10 +++++-----
 .../zydis/amalgamated-dist/Zydis.h             | 18 ++++++++++++++++++
 2 files changed, 23 insertions(+), 5 deletions(-)

diff --git a/subprojects/packagefiles/zydis/amalgamated-dist/Zydis.c b/subprojects/packagefiles/zydis/amalgamated-dist/Zydis.c
index 66dcb1ed3d..d088000762 100644
--- a/subprojects/packagefiles/zydis/amalgamated-dist/Zydis.c
+++ b/subprojects/packagefiles/zydis/amalgamated-dist/Zydis.c
@@ -54437,7 +54437,7 @@ static const char* const DECIMAL_LOOKUP =
 /* Decimal                                                                                        */
 /* ---------------------------------------------------------------------------------------------- */
 
-#if defined(ZYAN_X86) || defined(ZYAN_ARM) || defined(ZYAN_EMSCRIPTEN) || defined(ZYAN_WASM) || defined(ZYAN_PPC) || defined(ZYAN_S390)
+#if ZYAN_ARCHITECTURE_WIDTH != 64
 static ZyanStatus ZydisStringAppendDecU32(ZyanString* string, ZyanU32 value, ZyanU8 padding_length)
 {
     ZYAN_ASSERT(string);
@@ -54529,7 +54529,7 @@ static ZyanStatus ZydisStringAppendDecU64(ZyanString* string, ZyanU64 value, Zya
 /* Hexadecimal                                                                                    */
 /* ---------------------------------------------------------------------------------------------- */
 
-#if defined(ZYAN_X86) || defined(ZYAN_ARM) || defined(ZYAN_EMSCRIPTEN) || defined(ZYAN_WASM) || defined(ZYAN_PPC)
+#if ZYAN_ARCHITECTURE_WIDTH != 64
 static ZyanStatus ZydisStringAppendHexU32(ZyanString* string, ZyanU32 value, ZyanU8 padding_length,
     ZyanBool force_leading_number, ZyanBool uppercase)
 {
@@ -54694,7 +54694,7 @@ ZyanStatus ZydisStringAppendDecU(ZyanString* string, ZyanU64 value, ZyanU8 paddi
         ZYAN_CHECK(ZydisStringAppend(string, prefix));
     }
 
-#if defined(ZYAN_X64) || defined(ZYAN_AARCH64) || defined(ZYAN_PPC64) || defined(ZYAN_RISCV64) || defined(ZYAN_LOONGARCH) || defined(ZYAN_S390)
+#if ZYAN_ARCHITECTURE_WIDTH == 64
     ZYAN_CHECK(ZydisStringAppendDecU64(string, value, padding_length));
 #else
     if (value & 0xFFFFFFFF00000000)
@@ -54720,7 +54720,7 @@ ZyanStatus ZydisStringAppendHexU(ZyanString* string, ZyanU64 value, ZyanU8 paddi
         ZYAN_CHECK(ZydisStringAppend(string, prefix));
     }
 
-#if defined(ZYAN_X64) || defined(ZYAN_AARCH64) || defined(ZYAN_PPC64) || defined(ZYAN_RISCV64) || defined(ZYAN_LOONGARCH) || defined(ZYAN_S390)
+#if ZYAN_ARCHITECTURE_WIDTH == 64
     ZYAN_CHECK(ZydisStringAppendHexU64(string, value, padding_length, force_leading_number,
         uppercase));
 #else
@@ -55009,4 +55009,4 @@ ZyanStatus ZydisIsFeatureEnabled(ZydisFeature feature)
     }
 }
 
-/* ============================================================================================== */
\ No newline at end of file
+/* ============================================================================================== */
diff --git a/subprojects/packagefiles/zydis/amalgamated-dist/Zydis.h b/subprojects/packagefiles/zydis/amalgamated-dist/Zydis.h
index 83972e7320..33c67c8a78 100644
--- a/subprojects/packagefiles/zydis/amalgamated-dist/Zydis.h
+++ b/subprojects/packagefiles/zydis/amalgamated-dist/Zydis.h
@@ -207,27 +207,37 @@
 
 #if defined(_M_AMD64) || defined(__x86_64__)
 #   define ZYAN_X64
+#   define ZYAN_ARCHITECTURE_WIDTH 64
 #elif defined(_M_IX86) || defined(__i386__)
 #   define ZYAN_X86
+#   define ZYAN_ARCHITECTURE_WIDTH 32
 #elif defined(_M_ARM64) || defined(__aarch64__)
 #   define ZYAN_AARCH64
+#   define ZYAN_ARCHITECTURE_WIDTH 64
 #elif defined(_M_ARM) || defined(_M_ARMT) || defined(__arm__) || defined(__thumb__)
 #   define ZYAN_ARM
+#   define ZYAN_ARCHITECTURE_WIDTH 32
 #elif defined(__EMSCRIPTEN__) || defined(__wasm__) || defined(__WASM__)
 #   define ZYAN_WASM
+#   define ZYAN_ARCHITECTURE_WIDTH 32
 #elif defined(__loongarch__)
 #   define ZYAN_LOONGARCH
+#   define ZYAN_ARCHITECTURE_WIDTH 64
 #elif defined(__powerpc64__) || defined(__ppc64__) || defined(__PPC64__) || defined(__POWERPC64__) || \
     defined(_ARCH_PPC64)
 #   define ZYAN_PPC64
+#   define ZYAN_ARCHITECTURE_WIDTH 64
 #elif defined(__powerpc__) || defined(__powerpc) || defined(__ppc__) || defined(__PPC__) || \
     defined(__POWERPC__) || defined(_ARCH_PPC)
 #   define ZYAN_PPC
+#   define ZYAN_ARCHITECTURE_WIDTH 32
 #elif defined(__riscv) || defined(__riscv__)
 #   if __riscv_xlen == 64
 #       define ZYAN_RISCV64
+#       define ZYAN_ARCHITECTURE_WIDTH 64
 #   else
 #       define ZYAN_RISCV32
+#       define ZYAN_ARCHITECTURE_WIDTH 32
 #   endif
 #elif defined(__arc__)
 #   define ZYAN_ARC
@@ -241,6 +251,14 @@
 #   error "Unsupported architecture detected"
 #endif
 
+#if !defined(ZYAN_ARCHITECTURE_WIDTH)
+#   if defined(__LP64__)
+#       define ZYAN_ARCHITECTURE_WIDTH 64
+#   else
+#       define ZYAN_ARCHITECTURE_WIDTH 32
+#   endif
+#endif
+
 /* ============================================================================================== */
 /* Endianness detection                                                                           */
 /* ============================================================================================== */
-- 
2.55.0

